api/proto/external.proto

```diff
@@ -226,6 +226,20 @@ message TxIn {
 
     // Proof that each TxOut in `ring` is in the ledger.
     repeated TxOutMembershipProof proofs = 2;
+
+    // Any rules specified by the signed input
+    InputRules input_rules = 3;
+}
+
+// Rules enforced on a transaction by a signed input within it (MCIP #31)
+message InputRules {
+    // Outputs required to appear in the TxPrefix for the Tx to be valid
+    repeated TxOut required_outputs = 1;
+
+    // A maximum value which the tombstone block for the Tx cannot exceed
+    //
+    // A value of zero here means no limit is enforced
```

Comment:
Could this make use of proto's `optional` field? https://developers.google.com/protocol-buffers/docs/proto#optional. 

Might be clearer than using `0` to signify "not set".

Reply:
We are using `proto3` where the `optional` keyword has been deprecated. All fields in proto land are optional.

Reply:
But also, for things like integers, "optional" in proto3 means "defaults to zero" and the default cannot be adjusted. This is something we cannot change if we want to use proto3